### PR TITLE
Catch up main

### PR DESCRIPTION
**Pull request by andrewballantyne**
Resolves: #549 
Work was coded for the 2.1-hotfix: #532 

## Description

Couple gaps we needed to close for ODH in general and the RHODS infra.

## How Has This Been Tested?

Just testing the Notebook creation is all that is needed.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by lucferbux — APPROVED** [review]
/lgtm

File: frontend/src/pages/notebookController/screens/server/StartServerModal.tsx [hunk 1]
```diff
@@ -51,9 +53,8 @@ const StartServerModal: React.FC<StartServerModalProps> = ({ open, spawnInProgre
     }
   }, [open]);
 
-  const notebookLink = notebook?.metadata.annotations?.['opendatahub.io/link'];
-
-  const spawnFailed = spawnStatus?.status === AlertVariant.danger;
+  const spawnFailed =
+    spawnStatus?.status === AlertVariant.danger || spawnStatus?.status === AlertVariant.warning;
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/notebookController/screens/server/StartServerModal.tsx:57]: Could you tell me why do you add `warning` back to the spawn failed flag? The warning currently only happens when
```
case 'Killing': {
  currentEvent = 'Stopping container oauth-proxy';
  status = EventStatus.WARNING;
  break;
}
```
We don't want to change the `Cancel` button to `Close` which makes the user think the spawning failed in this situation.

Comment (andrewballantyne) [reply frontend/src/pages/notebookController/screens/server/StartServerModal.tsx:57]: Oops, bad conflict resolution. My bad. I read the diff wrong. Let me update.

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Pull request merged**